paddle/operators/sigmoid_op.cc

```diff
@@ -18,11 +18,10 @@ namespace operators {
 
 class SigmoidOp : public OperatorWithKernel {
 protected:
-  void InferShape(const std::vector<const Tensor *> &inputs,
-                  const std::vector<Tensor *> &outputs) const override {
-    PADDLE_ENFORCE(inputs.size() == 1, "Sigmoid Op only have one input");
-    PADDLE_ENFORCE(outputs.size() == 1, "Sigmoid Op only have one output");
-    outputs[0]->Resize(inputs[0]->dims());
+  void InferShape(const InferShapeContext &ctx) const override {
+    PADDLE_ENFORCE(ctx.InputSize() == 1, "Sigmoid Op only have one input");
```

Comment:
enforce the number of real inputs is the duty of `OperatorBase`, here `1` is just the number of inputs in definition of sigmoid.

The `simgoid`'s op definition has only one input `X`

```c++
  SigmoidOpMaker(OpProto *proto, OpAttrChecker *op_checker)
      : OpProtoAndCheckerMaker(proto, op_checker) {
    AddInput("X", "sigmoid input");
    AddOutput("Y", "sigmoid output");
    AddComment("Sigmoid function");
  }
```
so we need `OperatorBase`'s infershape to automatically enforce number of inputs and outputs, the real operator's implementation doesn't need to enforce itself.
@jacquesqiao @reyoung @QiJune